arcgis-ios-sdk-samples/Display information/Display grid/DisplayGridSettingsViewController.swift

```diff
@@ -241,77 +241,105 @@ class DisplayGridSettingsViewController: UITableViewController {
     // MARK: - UITableViewDelegate
     
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let cell = tableView.cellForRow(at: indexPath)
+        switch tableView.cellForRow(at: indexPath) {
+        case gridTypeCell:
+            showGridTypePicker()
+        case labelPositionCell:
+            showLabelPositionPicker()
+        case labelFormatCell:
+            showLabelFormatPicker()
+        case labelUnitCell:
+            showLabelUnitPicker()
+        case gridColorCell:
+            showGridColorPicker()
+        case labelColorCell:
+            showLabelColorPicker()
+        default:
+            break
+        }
+    }
+    
+    private func showLabelPositionPicker() {
         guard let grid = mapView?.grid else {
             return
         }
-        switch cell {
-        case gridTypeCell:
-            if let gridType = GridType(grid: grid) {
-                let selectedIndex = gridType.rawValue
-                let labels = GridType.allCases.map { (type) -> String in
-                    return type.label
-                }
-                let optionsViewController = OptionsTableViewController(labels: labels, selectedIndex: selectedIndex) { (newIndex) in
-                    self.changeGrid(to: GridType(rawValue: newIndex)!)
+        let selectedIndex = grid.labelPosition.rawValue
+        let optionsViewController = OptionsTableViewController(labels: labelPositionLabels, selectedIndex: selectedIndex) { (newIndex) in
+            self.mapView?.grid?.labelPosition = AGSGridLabelPosition(rawValue: newIndex)!
+        }
+        optionsViewController.title = "Position"
+        show(optionsViewController, sender: self)
+    }
+    
+    private func showLabelFormatPicker() {
+        if let grid = mapView?.grid,
```

Comment:
Would this make more sense as a guard statement?

Reply:
👍